components/dialog/RollbackWorkloadDialog.vue

```diff
@@ -0,0 +1,211 @@
+        this.close();
+      } catch (err) {
+        this.errors = exceptionToErrorsArray(err);
+      }
+    },
+    getRevisionNumber( revision ) {
+      return revision.metadata.annotations['deployment.kubernetes.io/revision'];
+    },
+    buildRevisionOption( revision ) {
+      const revisionNumber = this.getRevisionNumber(revision);
+      const isCurrentRevision = this.getRevisionNumber(revision) === this.currentRevisionNumber;
+      const now = day();
+      const createdDate = day(revision.metadata.creationTimestamp);
+      const revisionAge = diffFrom(createdDate, now, this.t);
+      const units = this.t(revisionAge.unitsKey, { count: revisionAge.label });
+      const currentLabel = this.t('promptRollback.currentLabel');
+      const optionLabel = this.t('promptRollback.revisionOption', {
+        revisionNumber,
+        revisionAge:    revisionAge.label,
+        units,
+        currentLabel:   isCurrentRevision ? currentLabel : ''
+      });
+
+      return {
+        label:    optionLabel,
+        value:    revision,
+        disabled: isCurrentRevision
+      };
+    },
+    getOptionLabel(option) {
+      return option.label;
+    },
+    fetchWorkloadRevisions() {
+      this.$store.dispatch('cluster/findAll', { type: WORKLOAD_TYPES.REPLICA_SET })
+        .then(( response ) => {
+          const allReplicaSets = response;
+
+          const hasRelationshipWithCurrentWorkload = ( replicaSet ) => {
+            const relationshipsOfReplicaSet = replicaSet.metadata.relationships;
+
+            const revisionsOfCurrentWorkload = relationshipsOfReplicaSet.filter(( relationship ) => {
+              const isRevisionOfCurrentWorkload = relationship.fromId && relationship.fromId === `${ this.workloadNamespace }/${ this.workloadName }`;
+
+              return isRevisionOfCurrentWorkload;
+            });
+
+            return revisionsOfCurrentWorkload.length > 0;
+          };
+
+          const workloadRevisions = allReplicaSets.filter(( replicaSet ) => {
+            return hasRelationshipWithCurrentWorkload( replicaSet );
+          });
+
+          const revisionOptions = workloadRevisions.map( (revision ) => {
+            const isCurrentRevision = this.getRevisionNumber(revision) === this.currentRevisionNumber;
+
+            if (isCurrentRevision) {
+              this.currentRevision = revision;
+            }
+
+            return this.buildRevisionOption( revision );
+          });
+
+          this.revisions = revisionOptions;
+        })
+        .catch(( err ) => {
+          this.errors = exceptionToErrorsArray(err);
+        });
+    }
+  },
+  created() {
+    this.fetchWorkloadRevisions();
+  },
+};
+</script>
+
+<template>
+  <Card
+    class="prompt-rollback"
+    :show-highlight-border="false"
+  >
+    <h4 slot="title" class="text-default-text">
+      {{ t('promptRollback.modalTitle', { workloadName }, true) }}
+    </h4>
+    <div slot="body" class="pl-10 pr-10">
+      <form>
+        <LabeledSelect
```

Comment:
If there is only 1 revision (the current one), perhaps we should show a banner under the select saying that there are no revisions to rollback to.

Reply:
updated